src/db_api_util.ts

```diff
@@ -145,19 +145,44 @@ export async function setThreadNeedsAttentionToDb(
   }
 }
 
+export async function setThreadHistoryTs(
+  slackParentMessageTs: string,
+  historyTs: string
+): Promise<void> {
+  const client = await pool.connect();
+  try {
+    await client.query(
+      'UPDATE threads SET history_ts = $1 WHERE slack_parent_message_ts = $2;',
+      [historyTs, slackParentMessageTs]
+    );
+  } catch (error) {
+    logger.info('Failed to update threads; ignoring for now!');
+  } finally {
+    client.release();
+  }
+}
+
 export async function getThreadLatestMessage(
   slackParentMessageTs: string
```

Comment:
The history_ts is the id of the 'chat history' message that opens up a thread.  It's used only if there are no other messages in the thread, because otherwise when you link to it slack will highlight the message in the channel and not open up a thread view.  This function will prefer to use the newest message in the chat, otherwise the history item as a fallback.

We could always link to the top of the chat (the history), but I think the latest message in the conversation is better!